### ISSUE TITLE
How do I contribute?

### ISSUE THREAD

**Issue by fresheneesz** [plotly/documentation#550]
This project is incredibly confusing. I found [this page](https://github.com/plotly/documentation/blob/source-design-merge/Contributing.md),  but it doesn't help me figure out how to edit what I want to edit. I want to improve the [documentation on annotations](https://plot.ly/javascript/reference/#layout-annotations). But the [folder that seems to be for that](https://github.com/plotly/documentation/tree/source-design-merge/_posts/plotly_js/annotations) doesn't seem to contain anything like what I see on that page.  So how do I make changes to that part of the documentation?

**Comment by cldougl** [comment]
Hi there,

The annotation folder generates [this documentation](https://plot.ly/javascript/text-and-annotations/).

The reference page is generated [here](https://github.com/plotly/documentation/blob/source-design-merge/_posts/2015-08-19-plotly_js-reference.html). The content for all reference pages is generated from the plotly.js attribute descriptions (for example: https://github.com/plotly/plotly.js/blob/appveyor/src/components/annotations/attributes.js) so to change that page you would have to make a PR in the [plotly.js](https://github.com/plotly/plotly.js) repo.

**Comment by fresheneesz** [comment]
Thanks, that's what I figured but I couldn't find it in that repo when i looked. Could this kind of info be added to the [Contributing.md](https://github.com/plotly/documentation/blob/source-design-merge/Contributing.md) page so people have a way to learn about that?

**Comment by fresheneesz** [comment]
Also, one of the things I want to do is to change the order that items under 'annotations' appear. How is that order determined?

**Comment by cldougl** [comment]
Typically, I think it's really great if users want to contribute to the examples in the docs! To me it makes more sense if the attributes are written by people writing the codebase/functions using those attributes. It's important that users can note when an attribute isn't described properly so I don't mind adding a note to the contributing.md about how that file is generated and point people in the direction to make a note about improving an attribute description.

**Comment by cldougl** [comment]
I'm not sure how the order is determined off the top of my head. There are plans to improve the reference page to make it much more user-friendly, and better ordering is included there, but I don't have a timeline.

**Comment by fresheneesz** [comment]
"it makes more sense if the attributes are written by people writing the codebase/functions using those attributes."

It makes sense unless they're not writing understandable docs. Changing the order would go a long way to help me understand annotations, since related properties are currently scattered around in there, for example ayref and axref are separated from ay and ax, so you have to scroll around. Also, there's tons of redundancy in this reference in general. the entire sections X Axis and Y Axis look identical. In the annotations section specifically, things like xref and yref could be combined, since they only difference is the axis they relate to. All this redundancy makes it much harder to ensure I know how to use this stuff. 

Where are the plans to improve the reference page and how can I help?

**Comment by cldougl** [comment]
yep by all means you can make prs to edit the attribute descriptions in the js repo!

**Comment by cldougl** [comment]
>Where are the plans to improve the reference page and how can I help?

There are a few issues logged in this repo at the moment that should be addressed. I've been prioritizing working on broken/missing docs and have not gotten to 'improving' yet - but if you'd like to take a crack at creating a logical order you're more than welcome to!!

**Comment by fresheneesz** [comment]
I think an order that makes more sense is:

* text
* font
* align
* bgcolor
* opacity
* arrowsize
* textangle
* borderwidth
* bordercolor
* borderpad
* showarrow
* arrowcolor
* arrowhead
* arrowwidth
* y 
* x
* xref
* yref
* ay
* ax
* axref
* ayref
* xanchor
* yanchor

**Comment by cldougl** [comment]
I'd probably be more consistent with ordering x and y:
>y
x
xref
yref
ay
ax
axref
ayref
xanchor
yanchor

It may also help if you add a small explanation on why you think that order makes more sense, at the moment I disagree. Looks like `arrowsize` is towards the top, before `showarrow` and other arrow attributes. Additionally, the `x / y / xref / yref` attributes are lower down, but are used more frequently than many of the stylistic attributes.

**Comment by fresheneesz** [comment]
Yes, good points. How about this:

* text
* font
* align
* bgcolor
* opacity
* textangle
* x
* y 
* xref
* yref
* ax
* ay
* axref
* ayref
* xanchor
* yanchor
* borderwidth
* bordercolor
* borderpad
* showarrow
* arrowsize
* arrowcolor
* arrowhead
* arrowwidth

Text is at the top since its required. I put the text styling attributes right under text, because they relate conceptually to that text. All the x/y/ax/ay things should be kept together because they're all positioning related. Border related stuff is grouped and arrow styling related stuff are grouped, but are below positioning because they're used less often. Sound good?
